#36 마일리지 수정

0hun opened the pull request.

0hun committed:
#36 마일리지 추가 메소드 추가

관리자가 특정 회원의 마일리지를 추가하도록하는 메소드 추가

0hun committed:
#36 init.sql 수정

전화번호 100글자는 경우의수에 비해 크므로 30글자로 변경

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/mileageController.java:
```diff
@@ -44,5 +53,29 @@
+   * @return ResponseEntity
+   */
+  @LoginCheck(userLevel = UserLevel.ADMIN)
+  @PostMapping("/account")
```
여기서는 부분 수정으로 보이는데요~ POST가 맞을까요?

f-lab-dev replied:
그리고 역시 자신의 정보를 수정하는 Controller를 따로 빼는게 좋아보이네요

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/mileageController.java:
```diff
@@ -44,5 +53,29 @@
+    boolean existAccount = accountService.existsById(accountId);
+
+    if (!existAccount) {
+      return ACCOUNT_BADE_REQUEST;
```
`BADE` 오타로 보입니다~

0hun committed:
Merge branch 'issue/35' into issue/36

0hun committed:
#36 리뷰 반영

관리자가 사용자의 마일리지를 수정하는 메소드 추가

0hun committed:
#36 마일리지 수정

마일리지가 존재하지 않을 경우 마일리지 추가
존재할경우 마일리리 수정

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/mileageController.java:
```diff
@@ -81,7 +82,13 @@
-    mileageService.updateMileage(mileage);
+    boolean existMileage = mileageService.existsByAccountId(accountId);
+
+    if (existMileage) {
```
이런 upsert는 쿼리로 구현이 가능합니다~ 한번 찾아보시면 좋겠네요

0hun committed:
#36 마일리지 수정

마일리지 수정시 upsert 적용하여
마일리지가 accoun_id가 이미 존재한다면 update 없다면 insert

upsert로 적용하여 로직을 줄이도록 한다.
마일리지 account_id unique 인덱스 적용하여 중복되지 않도록 한다.

0hun committed:
#36 마일리지 수정

클래스 파일 대문자로 시작하도록 수정
